### ISSUE TITLE
Geometry Clipping Box

### ISSUE THREAD

**Issue by Gorgious56** [IfcOpenShell/IfcOpenShell#4420]
With the new awesome clip planes features I can see a new feature which would be awesome. Instead of managing clipping planes we would have a box which would selectively update 8 clipping planes to show and hide geometry dynamically.

Similarly to the feature in PCV addon

![image](https://github.com/IfcOpenShell/IfcOpenShell/assets/25156105/8dda346d-4297-4f21-a398-f20bacb68ec3)

![animation](https://github.com/IfcOpenShell/IfcOpenShell/assets/25156105/ae7bf386-77d2-4f5c-894f-17c7ba4554d7)

It could work with a physical elongated 3D cube, the nice thing it wouldn't need special gizmos to tweak since the transforms of the cube can be used to position the clipping planes.

FWIW here's how it would look (I used a boolean intersect modifier but the result would be the same-ish ?)

![animation](https://github.com/IfcOpenShell/IfcOpenShell/assets/25156105/725edadb-2057-4574-a95a-1a8a606e40ae)

Well the boolean has some problems with geometry but you see the point. I'd be willing to take a stab but unfortunately not in the foreseeable future since I'm a bit swamped atm. I juste wanted to test the waters beforehand and see if other people would like this feature, or maybe if someone want to chime in. Or maybe it's already in the works ?

**Comment by Andrej730** [comment]
Yeah, this definitely is the intention to build something similar as it's awkward using multiple clipping planes to build this kind of bounding box manually. Haven't started working on it yet.

**Comment by Moult** [comment]
Another intention is to build:

1. When a drawing is activated, two (invisible) clipping planes are added for the near and far clip.
2. An upgrade to the spatial manager, which allows you to isolate a storey which includes 2x clipping plane of the storey (at a few configurations, either below + at 1m, or 1m + at top for RCP, etc), or a clipping box of a space.

**Comment by Gorgious56** [comment]
Great to hear it's on the roadmap !
FWIW here's a new PCV addon implementation with custom gizmos. I've yet to test it but it looks slick.

https://blenderartists.org/t/point-cloud-visualizer/684180/547?u=gorgious

Cheers

**Comment by Moult** [comment]
That's incredible, I'd love to reuse their gizmo code and build it exactly as they have!

**Comment by ppaawweeuu** [comment]
@Moult , PCV is amazing tool, but what we see in @Gorgious56 link is preview mode of pointcloud. That cropping planse clips gizmo-points, before converting into real geometry. Gizmos clip gizmos.

**Comment by ppaawweeuu** [comment]
ough.. I just realized that there is new Explore Tool in BBIM. Dion, you did it! Awesome! Thanks! :)
Btw, if you looking for some gizmo & beautiful UI witchcraft - see MASHIN3 yt channel: https://www.youtube.com/@MACHIN3

**Comment by BIMvoice** [comment]
This is resolved. @Gorgious56 already built exactly this: a box gizmo (Blender empty) whose transform drives 6 live clip planes, plus cross-section cap rendering, presets, and edit-mode resize handles, across PR #8178, #8185, #8191, #8192, and #8221 (all merged). Closing as shipped. Feel free to reopen if there's a specific gap not covered by the clip box feature.
